crates/tauri-utils/src/config.rs

```diff
@@ -1773,6 +1774,358 @@ pub struct AssetProtocolConfig {
+pub enum HeaderSource {
+  /// string version of the header Value
+  Inline(String),
+  /// list version of the header value. Item are joined by "," for the real header value
+  List(Vec<String>),
+  /// (Rust struct | Json | JavaScript Object) equivalent of the header value. Items are composed from: key + space + value. Item are then joined by ";" for the real header value
+  Map(HashMap<String, String>),
+}
+
+impl Display for HeaderSource {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    match self {
+      Self::Inline(s) => write!(f, "{s}"),
+      Self::List(l) => write!(f, "{}", l.join(", ")),
+      Self::Map(m) => {
+        let len = m.len();
+        let mut i = 0;
+        for (key, value) in m {
+          write!(f, "{} {}", key, value)?;
+          i += 1;
+          if i != len {
+            write!(f, "; ")?;
+          }
+        }
+        Ok(())
+      }
+    }
+  }
+}
+
+impl From<HeaderSource> for Vec<String> {
+  fn from(val: HeaderSource) -> Self {
+    match val {
+      HeaderSource::Inline(s) => {
+        let mut out: Vec<String> = Vec::new();
+        let mut separator = ',';
+        if s.contains(';') {
+          separator = ';';
+        }
+        for item in s.split(separator) {
+          out.push(String::from(item));
+        }
+        out
+      }
+      HeaderSource::List(l) => l,
+      HeaderSource::Map(m) => {
+        let mut out: Vec<String> = Vec::new();
+        for (key, value) in m.iter() {
+          let mut item = String::new();
+          item.push_str(key);
+          item.push(' ');
+          item.push_str(value);
+          out.push(item);
+        }
+        out
+      }
+    }
+  }
+}
+
+impl From<HeaderSource> for HashMap<String, String> {
+  fn from(val: HeaderSource) -> Self {
+    match val {
+      HeaderSource::Inline(s) => {
+        let mut out: HashMap<String, String> = HashMap::new();
+        for item in s.split(';') {
+          let index = item.find(' ');
+          if index.is_some() {
+            let (key, value) = item.split_at(index.unwrap());
+            out.insert(String::from(key), String::from(value));
+          }
+        }
+        out
+      }
+      HeaderSource::List(l) => {
+        let mut out: HashMap<String, String> = HashMap::new();
+        for item in l.iter() {
+          let index = item.find(' ');
+          if index.is_some() {
+            let (key, value) = item.split_at(index.unwrap());
+            out.insert(String::from(key), String::from(value));
+          }
+        }
+        out
+      }
+      HeaderSource::Map(m) => m,
+    }
+  }
+}
+
+impl From<String> for HeaderSource {
+  fn from(value: String) -> Self {
+    Self::Inline(value)
+  }
+}
+
+impl From<Vec<String>> for HeaderSource {
+  fn from(value: Vec<String>) -> Self {
+    Self::List(value)
+  }
+}
+
+impl From<HashMap<String, String>> for HeaderSource {
+  fn from(value: HashMap<String, String>) -> Self {
+    Self::Map(value)
+  }
+}
```

Comment:
I don't see this implementations used anywhere, right? we can remove them in that case